commit 8ae09b6f4afa8ceb2770da67b6dc79c78b86af47
Author: David Faure <faure@kde.org>
Date:   Mon Dec 19 13:29:30 2016 +0100

    Fix DATA LOSS bug in ical resource which failed to create std.ics if it didn't exist.
    
    The default setup sets the Path to be a local path, not a URL.
    => Use QUrl::fromUserInput so that it can deal with both cases, paths and URLs.
    
    CCMAIL: smartins@kde.org, release-team@kde.org

diff --git a/resources/shared/singlefileresource/singlefileresource.h b/resources/shared/singlefileresource/singlefileresource.h
index f3c19aaf5..3f4149903 100644
--- a/resources/shared/singlefileresource/singlefileresource.h
+++ b/resources/shared/singlefileresource/singlefileresource.h
@@ -55,7 +55,7 @@ public:
         , mSettings(new Settings(config()))
     {
         // The resource needs network when the path refers to a non local file.
-        setNeedsNetwork(!QUrl(mSettings->path()).isLocalFile());
+        setNeedsNetwork(!QUrl::fromUserInput(mSettings->path()).isLocalFile());
     }
     ~SingleFileResource()
     {
@@ -82,7 +82,7 @@ public:
             return;
         }
 
-        mCurrentUrl = QUrl(mSettings->path()); // path already has scheme
+        mCurrentUrl = QUrl::fromUserInput(mSettings->path()); // the string contains the scheme if remote, doesn't if local path
         if (mCurrentHash.isEmpty())
         {
             // First call to readFile() lets see if there is a hash stored in a
